docs/manual/philosophy.md

```diff
@@ -0,0 +1,149 @@
+
+Libraries should be packaged separately and binaries should link to
+the version provided by system packages. Static linking is
+discouraged. This limits the number of packages that need updates or
+re-builds in case of a vulnerability.
+
+#### Unattended operation
+
+Package installation and update is unattended and must not require
+user interaction. When possible delay user interaction to the first
+start up.
+
+#### Clear update path
+
+Each package name creates an update path where packages with the same
+name and a newer version (technically Epoch-Version-Release) are an
+update for the packages with lower version. Packages are not allowed
+to make assumptions on what intermediate packages were installed or
+not.
+
+#### Build Sources, Changes and Build instructions together
+
+Source packages bundle upstream sources, patches and build
+instructions into one entity. If changes need to be made everything
+needed is available, including a list of packages needed to run the
+build.
+
+Nosource packages are a exception that does not ship the sources but
+only the build instructions to keep proprietary source private. They
+still can be used to generate binary packages in combination with the
+sources.
+
+#### Separate Upstream Source from Patches
+
+Source packages are supposed to contain the unaltered upstream sources
+in a way that their checksum can be checked. All changes can be done
+by applying patches or running build scripts. This makes it easy to
+understand what the packager did to the software. This is important to
+be able to figure out where issues arise from and to see which issues
+are fixed in the package - even if the upstream version is still
+vulnerable.
+
+#### Support back porting fixes
+
+A distribution often has a different approach to fixing software from an
+upstream project. Upstream projects are often content with fixing
+their latest release(s). Distribution often support a particular
+version of each software and need those to be fixed. The Sources and
+Patches approach makes handling this a lot easier (although it is
+lately being extended with the use of version control like git). 
+
+### Allow 3rd party packages
+
+Although RPM is designed to package a whole distribution it explicitly
+supports 3rd parties to also provide packages without becoming part of
+the distribution and their processes.
+
+Rpmbuild can be run locally without the use of a big build
+system. Dependencies are in large part automatic and target libraries
+inside the packages instead of relying on conventions like package
+names. This way 3rd party packages can be checked for compatibility
+beyond them just claiming that they target the distribution in question.
+
+## Separating Upstream Sources and Packager's Changes
+
+Sources are given as files and patches. Upstream tarballs should be
+used unchanged and patched during the build. This way packages can be
+reviewed more easily and the changes made to the upstream project are
+explicit.
```

Comment:
The rpm lore term for this is "pristine sources", I'd stick it somewhere in there :sweat_smile:

Reply:
Also this is practically the same as "Separate Upstream Source from Patches" a couple of sections earlier. I get that there's a slight difference in perspective, but  its quite repetitive. Backporting fixies too. 

Maybe have a single "Pristine sources" section for the main beef, and elaborate on these things it enables as subsections?